### PR TITLE
feat: Banner image placeholder and fallback background

### LINKED ISSUE #2925: Improve landing page performance
We have an Image service that gives the Telescope landing page a random picture on page load. This is nice but not yet optimal. 
On slow internet connections, these high-quality banner images take a couple of seconds to load and reduce the overall performance.

Moving forward, we should improve the Image service and the front page to: 
 - Return [responsive image ](https://developer.mozilla.org/en-US/docs/Learn/HTML/Multimedia_and_embedding/Responsive_images), with various sizes that fit well on different devices
 - Return a low-quality, blurred-out image for initial page load and replace it with a high definition one when it finishes loading

**Why would you like this to be added**:

### PR DESCRIPTION
**Pull request by DukeManh**
## Issue This PR Addresses
Fixes #2925

- Add loading banner image background by displaying a tiny 40px wide image (minimal loading time) and transitioning  in the main 2000px wide image (long loading time) when it finishes loading 
- Add fallback gradient banner background
- Fix inconsistent image src, request a specific photo versus a random image chosen by the backend

## Test
- Pull this change locally
- Run server: 
 ```
    cd src/api/image && pnpm dev

    // in another terminal
    cp config/env.staging .env

    // change IMAGE_URL in .env
    # Image Service URL
    IMAGE_URL=http://localhost:4444

   // Run the frontend
   pnpm dev
```
- to see that a gradient background is shown first, followed by a placeholder image, and then the high resolution image.
- If you are not able to see the difference, your internet is probably too fast, good. Go to the Network tab, click `disable cache`, and choose one of the `throttling` options.

## Screenshot

https://user-images.githubusercontent.com/51073515/158033934-5b8e2ca1-6a07-43be-9c0f-ba5c27bb7fa6.mp4

### CODE REVIEW

File: src/api/image/src/routes/image.js
Comment (humphd) [inline src/api/image/src/routes/image.js:22]: The comment above this line needs to be updated to include the fact that we now allow passing an integer too.

```
* We also support passing an image name as a param in the URL:
 *
 * - image: should look like look '_ok8uVzL2gI.jpg'. Don't allow filenames like '../../dangerous/path/traversal'.
 */
```

File: src/api/image/src/routes/image.js
Comment (humphd) [inline src/api/image/src/routes/image.js:85]: Let's throw away fractional components so we have a pure integer: `getPhotoAt(image|0);`

File: src/web/src/components/BannerDynamicItems.tsx
Comment (humphd) [inline src/web/src/components/BannerDynamicItems.tsx:22]: Should we use a larger upper-bound here so that we can access images beyond 100 when we get there?  Your circular buffer should be fine with something like 1000.

**Review by humphd — CHANGES_REQUESTED** [review]
So cool, I love it.  A few things I notice.

**Comment by DukeManh** [comment]
@humphd, I'm running this again and the images fail to load. Do you know what this error is?

image

**Comment by humphd** [comment]
I'm not 100% sure.  Is that Firefox?  Do you have a Service Worker on localhost:4444 that's interfering somehow (try changing port)?

File: src/api/image/src/routes/image.js
Comment (DukeManh) [reply src/api/image/src/routes/image.js:85]: Thanks, I forgot about this.

**Comment by DukeManh** [comment]
> I'm not 100% sure. Is that Firefox? Do you have a Service Worker on localhost:4444 that's interfering somehow (try changing port)?

It's on any browser I try.

Looks like something was changed on master recently that make the image service have CORS issue. I rebased and have the issue.

I filed #3212

**Comment by DukeManh** [comment]
This is currently blocked by #3212, which makes it hard to test.

File: src/web/src/components/DynamicImage.tsx
Comment (humphd) [inline src/web/src/components/DynamicImage.tsx:62]: This is a nice touch.

**Review by humphd — APPROVED** [review]

**Comment by rclee91** [comment]
I'm wondering if we might need to come to a consensus if we're ignoring Eslint tests until #3228 has been merged. 
Do we approve if all other tests pass

**Comment by humphd** [comment]
@rclee91 I think that's fine, and we can fix things that we catch when the checks are working properly.

**Review by rclee91 — APPROVED** [review]

**Comment by DukeManh** [comment]
It's interesting the ESLint recommends `Number.isNaN` instead of `isNaN`, but they behave a bit differently. https://stackoverflow.com/questions/33164725/confusion-between-isnan-and-number-isnan-in-javascript

**Review by humphd — DISMISSED** [review]

**Review by humphd — APPROVED** [review]

**Pull request merged**
